app/internal/agenda_events.py

```diff
@@ -59,11 +59,40 @@ def get_time_delta_string(start: date, end: date) -> str:
 
 
 def filter_dates(
-        events: List[Event], start: Optional[date],
-        end: Optional[date]) -> Iterator[Event]:
-    """filter events by a time frame."""
+        events: List[Event],
+        start: Union[None, date] = None,
+        end: Union[None, date] = None,
+) -> Iterator[Event]:
+    """Returns all events in a time frame.
+    if "start_date" or "end_date" are None,
+    it will ignore that parameter when filtering.
 
-    yield from (
-        event for event in events
-        if start <= event.start.date() <= end
-    )
+    for example:
+    if start_date = None and end_date = datetime.now().date,
+    then the function will return all events that ends before end_date."""
+
+    for event in events:
+        if start and end:
```

Comment:
Outside the for, define:

```python
start = start or datetime.date.min
end = end or datetime.date.max
```

Inside the for, use:
```python
if start <= event.start.date() <= end:
    yield event
```

Reply:
Great advice, thanks!